### DISCUSSION TITLE
Have python version update managed by renovate (not python dependencies)

### DISCUSSION THREAD

**Discussion by shadycuz** [renovatebot/renovate#22222] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

I have projects that depend on specific versions of Python. Like python `3.10.X`, `3.9.X`, `3.8.X` etc. I use `pyenv` to specify which versions of python/s my project requires. This happens using the [`.python-version`](https://github.com/pyenv/pyenv/tree/806b30d6ce5b263a765648fbcdd68266833b7289#choosing-the-python-version) file. 

The problem is when a new version of Python is updated. I have to manually change all these files, across all of my projects. It would be nice to have Renovate handle this for me. 

One example would be if I had a `.python-version` file of:
```
3.10.4.
```

I would expect renovate to change it to:
```
3.10.5
```

It would also be nice if Renovate would remove versions when they were deprecated and add new versions of Python when they were generally available.

**Comment by rarkins** [comment]
There might be multiple places where python versions are defined - including within tools like poetry and pipenv. To update them all:
- Renovate should recognize and be able to support them all
- Updates to Python should be grouped together so they're updated at the same time

If Renovate _was_ updating such versions, in which scenario would you end up with a deprecated version? e.g. I can imagine that happening if the project ignores Renovate's PRs for years, but that doesn't seem like a useful scenario

**Reply by shadycuz** [reply]
Well supporting them all is very ambitious and I would be happy if you supported just my use case first =). 

> If Renovate was updating such versions, in which scenario would you end up with a deprecated version?

Imagine I have a python version file with one or more versions:
```
3.7.16 3.8.16 3.9.16
```

So in 1 month and two weeks, when `3.7.16` becomes EOL. I would expect Renovate to remove it from the `.python-versions` file.

```
3.8.16 3.9.16
```

I'm not familiar with the renovate code base but if its opensource, I might be able to take a shot at implementing this feature for the `.python-versions` use case. I'm not sure where we would source the data from. https://endoflife.date/python seems to have an API and it looks to be free.

**Reply by rarkins** [reply]
Removing or adding versions is currently out of scope - Renovate does not have an elegant way to do that. Removing would probably be easier than adding.

The first step would be adding a `pyenv` manager to Renovate - I created a feature request here: #22225
